crates/mun_abi/src/autogen_impl.rs

```diff
@@ -162,6 +162,13 @@ impl StructInfo {
                 )
             })
     }
+
+    /// Returns the size of the struct
+    pub fn size(&self) -> usize {
```

Comment:
Note that this might not be right, it could be that for cache efficiency there are padding bytes.

Reply:
If we take a struct:
```struct Foo {
  x: i16,
  y: u64
  _padding: [u8; 4],
}```

Both for the packed and unpacked version: the offset and size would be stored for `padding`, making the `size` function valid.

Reply:
I assume that LLVM also has something to say about the size of the struct regardless of whether you pad it yourself or not. But Im really not sure.

Reply:
Hmm. That would just be an offset between array elements, right? That doesn't affect the actual size of the type?
We could write a test for this, though - to be sure.

Reply:
Mmm maybe its only the case for alignment

```
struct A {
  a: u16,
  b: u8,
}
```

Would probably have an alignment of `u32`. WDYT?

Reply:
Yes, that's what I thought